### PR TITLE
Don't require token if enableAccessToken is disabled and the resource…

### PR DESCRIPTION
**Pull request by jgraff2**
… is not write protected.

### CODE REVIEW

File: api/v1/helpers/verbs/utils.js [hunk 1]
```diff
@@ -182,18 +182,25 @@ function isWritable(req, modelInst) {
...
       )
       .catch((err) => reject(err));
     } else {
-      reject(new apiErrors.ForbiddenError());
+      // check if isWritable by default (no writers)
+      modelInst.isWritableBy()
+      .then((ok) => ok ? resolve(modelInst) :
+        reject(new apiErrors.ForbiddenError('Resource is write protected'))
+      )
+      .catch((err) => reject(err));
```
Comment (iamigo) [inline api/v1/helpers/verbs/utils.js]: `catch(reject)` should work

File: api/v1/helpers/verbs/utils.js [hunk 2]
```diff
@@ -182,18 +182,25 @@ function isWritable(req, modelInst) {
...
       .then((ok) => ok ? resolve(modelInst) :
-          reject(new apiErrors.ForbiddenError())
+        reject(new apiErrors.ForbiddenError(
+          'Resource not writable by this user'))
       )
       .catch((err) => reject(err));
     } else {
-      reject(new apiErrors.ForbiddenError());
+      // check if isWritable by default (no writers)
+      modelInst.isWritableBy()
```
Comment (iamigo) [inline api/v1/helpers/verbs/utils.js]: I don't understand this part... what does isWritableBy return when you don't pass a username or userid argument?

Comment (jgraff2) [inline api/v1/helpers/verbs/utils.js]: It will return true if there are no writers, false if there are.

Comment (iamigo) [inline api/v1/helpers/verbs/utils.js]: Can you expand the comment here to describe this?

File: api/v1/helpers/verbs/utils.js [hunk 3]
```diff
@@ -182,18 +182,27 @@ function isWritable(req, modelInst) {
...
+        reject(new apiErrors.ForbiddenError(
+          'Resource not writable by this user'))
       )
-      .catch((err) => reject(err));
+      .catch(reject);
     } else {
-      reject(new apiErrors.ForbiddenError());
+      // check if isWritable with no user
+      // when not passed a user, isWritable will return true if
+      // the resource is not write protected, false if it is
```
Comment (shriramshankar) [inline api/v1/helpers/verbs/utils.js]: I think we want to use 
`/*
*
*/`  for multi-line comments instead of `//`. We can talk care of this later.

**Pull request merged**